bliss/catalog.py

```diff
@@ -10,32 +10,33 @@
 from torch import Tensor
 
 
+# old function, to be deleted
 def convert_mag_to_nmgy(mag):
     return 10 ** ((22.5 - mag) / 2.5)
 
 
+# old function, to be deleted
 def convert_nmgy_to_mag(nmgy):
     return 22.5 - 2.5 * torch.log10(nmgy)
 
 
-def convert_nmgy_to_njymag(nmgy):
-    """Convert from flux (nano-maggie) to mag (nano-jansky), which is the format used by DC2.
+def convert_flux_to_magnitude(flux, c):
+    """Convert from flux to magnitude.
 
-    For the difference between mag (Pogson magnitude) and njymag (AB magnitude), please view
-    the "Flux units: maggies and nanomaggies" part of
-    https://www.sdss3.org/dr8/algorithms/magnitudes.php#nmgy
-    When we change the standard source to AB sources, we need to do the conversion
-    described in "2.10 AB magnitudes" at
+    For DC2, please set c to 3631.
+    The conversion is described in "2.10 AB magnitudes" at
     https://pstn-001.lsst.io/fluxunits.pdf
 
+
     Args:
-        nmgy: the fluxes in nanomaggies
+        flux: the flux in any linear unit
+        c: you may change this according to your survey dataset;
 
     Returns:
-        Tensor indicating fluxes in AB magnitude
+        Tensor indicating fluxes in magnitude
     """
 
-    return 22.5 - 2.5 * torch.log10(nmgy / 3631)
+    return 22.5 - 2.5 * torch.log10(flux / c)
```

Comment:
Better to subtract log10(c) for stability

Reply:
remove the 22.5 and rename `c` to `zero_point`